src/riak_kv_leveled_backend.erl

```diff
@@ -635,6 +637,31 @@ callback(Ref, UnexpectedCallback, State) ->
 %% Internal functions
 %% ===================================================================
 
+
+-spec log_fragmentation(eheap_alloc|binary_alloc) -> ok.
+log_fragmentation(Allocator) ->
+    FragStats =
+        lists:filter(
+            fun(MemStuff) ->
+                element(1, element(1, MemStuff)) == Allocator
```

Comment:
This is a bit risky.. assume data type changes in the future, then this may actually crash as a side effect of logging.
Rather make 3 match cases:
lists:filter(fun({{A, _}, _Stats}) -> A == Allocator;
                       (_) -> false
                       end,

Reply:
alternatively, build the new list straigth away:
FragStats = [ Stats || {{A, _}, Stats} <- econ_alloc:fragmentation(current), A == Allocator ]